src/config/mod.rs

```diff
@@ -281,20 +308,32 @@ fn parse_config(path: &std::path::Path) -> Result<(Vec<IpAddr>, Vec<CtrEntry>),
             ));
         }
 
-        let dns_servers: Option<Vec<IpAddr>> = if parts.len() == 5 && !parts[4].is_empty() {
-            let dns_server = match parts[4].split(',').map(|i| i.parse()).collect() {
-                Ok(i) => i,
-                Err(e) => {
-                    return Err(std::io::Error::new(
-                        std::io::ErrorKind::Other,
-                        format!("error parsing DNS server address {}: {}", parts[4], e),
-                    ))
+        let dns_servers: Option<Vec<IpAddr>> =
+            if parts.len() == 5 && !parts[4].is_empty() || !network_dns_servers.is_empty() {
+                let mut dns_server: Vec<IpAddr> = Vec::new();
+                if parts.len() == 5 && !parts[4].is_empty() {
+                    let ctr_dns_server: Vec<IpAddr> =
+                        match parts[4].split(',').map(|i| i.parse()).collect() {
+                            Ok(i) => i,
+                            Err(e) => {
+                                return Err(std::io::Error::new(
+                                    std::io::ErrorKind::Other,
+                                    format!("error parsing DNS server address {}: {}", parts[4], e),
+                                ))
+                            }
+                        };
+                    dns_server.extend(&ctr_dns_server);
                 }
+                // If network dns servers were specified then all the
+                // containers in this specified network must inherit
+                // network dns servers.
+                if !network_dns_servers.is_empty() {
+                    dns_server.extend(&network_dns_servers);
+                }
```

Comment:
Doesn't this just duplicate all network dns addresses for each container entry? This looks extremely inefficient. And I don't see how this can work with the next network update changes.

Reply:
@Luap99 Yes we can optimize this further if add a new mapping in backend i'd like to do that later on but can do it now as well if that's a concern.

> And I don't see how this can work with the next network update changes.

Entire backend is shared with all the threads so its recreated from the config on this disk on every SIGHUP. So on network update `netavark` will just rewrite the `config` file and send `SIGHUP` to aardvark it will do the needful. But I'll verify refresh model once again manually.

Reply:
> Yes we can optimize this further.

Will do it in the next push.

Reply:
Done.